crates/ruma-client-api/src/session/login.rs

```diff
@@ -226,22 +226,33 @@ pub mod v3 {
     #[serde(tag = "type", rename = "m.login.password")]
     pub struct Password {
         /// Identification information for the user.
-        pub identifier: UserIdentifier,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub identifier: Option<UserIdentifier>,
 
         /// The password.
         pub password: String,
+
+        /// Username for the user.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[deprecated = "\
+            Since Matrix Client-Server API r0.4.0, clients should instead use \
+            identifiers (`UserIdentifier`) instead of user.\
+        "]
+        pub user: Option<String>,
     }
 
     impl Password {
         /// Creates a new `Password` with the given identifier and password.
-        pub fn new(identifier: UserIdentifier, password: String) -> Self {
-            Self { identifier, password }
+        #[allow(deprecated)]
+        pub fn new(identifier: Option<UserIdentifier>, password: String) -> Self {
+            Self { identifier, password, user: None }
         }
     }
 
     impl fmt::Debug for Password {
+        #[allow(deprecated)]
         fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-            let Self { identifier, password: _ } = self;
+            let Self { identifier, password: _, user: _ } = self;
```

Comment:
We should print the user field in debug format

Reply:
Think I did this right, let me know if it isn't.

Reply:
This is good